datalad/runner/nonasyncrunner.py

```diff
@@ -552,21 +552,25 @@ def _handle_source_timeout(self, source):
         else:
             self._handle_file_timeout(source)
 
-    def _update_timeouts(self):
+    def _update_timeouts(self) -> bool:
         last_touched = list(self.last_touched.items())
         new_times = dict()
         current_time = time.time()
+        timeout_occurred = False
         for source, last_time in last_touched:
             if current_time - last_time >= self.timeout:
                 new_times[source] = current_time
                 self._handle_source_timeout(source)
+                timeout_occurred = True
         self.last_touched = {
             **self.last_touched,
             **new_times}
+        return timeout_occurred
 
-    def process_timeouts(self):
+    def process_timeouts(self) -> bool:
```

Comment:
I added a docstring in 1f9e74e55d7e39acf2516418a245b594433929ea. I would not like to change the name to `has_timedout`, because that sounds like a check for timeouts. The method `process_timeouts()` does a lot more, it updates the timeout-status, i.e. setting the runner up to receive new timeouts, and calls timeout handlers.